# Data processing of the 10 Years SPX daily returns:

In [6]:
import pandas as pd
import warnings 
warnings.filterwarnings("ignore")
data = pd.read_csv("Data/SPX_daily_returns_10Y.csv")

In [7]:
data.head(2)

,Unnamed: 0,A UN Equity,AAPL UW Equity,ABBV UN Equity,ABNB UW Equity,ABT UN Equity,ACGL UW Equity,ACN UN Equity,ADBE UW Equity,ADI UW Equity,...,WY UN Equity,WYNN UW Equity,XEL UW Equity,XOM UN Equity,XYL UN Equity,XYZ UN Equity,YUM UN Equity,ZBH UN Equity,ZBRA UW Equity,ZTS UN Equity
0,DATES,"day_to_day_total_return(start=-10Y,end=0D)","day_to_day_total_return(start=-10Y,end=0D)","day_to_day_total_return(start=-10Y,end=0D)","day_to_day_total_return(start=-10Y,end=0D)","day_to_day_total_return(start=-10Y,end=0D)","day_to_day_total_return(start=-10Y,end=0D)","day_to_day_total_return(start=-10Y,end=0D)","day_to_day_total_return(start=-10Y,end=0D)","day_to_day_total_return(start=-10Y,end=0D)",...,"day_to_day_total_return(start=-10Y,end=0D)","day_to_day_total_return(start=-10Y,end=0D)","day_to_day_total_return(start=-10Y,end=0D)","day_to_day_total_return(start=-10Y,end=0D)","day_to_day_total_return(start=-10Y,end=0D)","day_to_day_total_return(start=-10Y,end=0D)","day_to_day_total_return(start=-10Y,end=0D)","day_to_day_total_return(start=-10Y,end=0D)","day_to_day_total_return(start=-10Y,end=0D)","day_to_day_total_return(start=-10Y,end=0D)"
1,3/9/2016,0.001050144,0.000890825,0.01934201,NaN,0.002808272,-0.00116144,0.007600858,0.010906935,-0.007408746,...,0.002910149,0.016070761,NaN,-0.002783493,0.006330783,0.047909408,-0.00495566,-0.001690701,0.024555328,0.005717127


In [8]:
data.columns = [s.split()[0] for s in data.columns]
data = data.rename(columns = {"Unnamed:" : "Date"})
data = data.iloc[1:,:]
data["Date"] = pd.to_datetime(data["Date"], format = "%m/%d/%Y")
data.set_index("Date",inplace = True,drop=True)
data.head(2)

,A,AAPL,ABBV,ABNB,ABT,ACGL,ACN,ADBE,ADI,ADM,...,WY,WYNN,XEL,XOM,XYL,XYZ,YUM,ZBH,ZBRA,ZTS
Date,,,,,,,,,,,,,,,,,,,,,
2016-03-09,0.001050144,0.000890825,0.01934201,NaN,0.002808272,-0.00116144,0.007600858,0.010906935,-0.007408746,0.000832639,...,0.002910149,0.016070761,NaN,-0.002783493,0.006330783,0.047909408,-0.00495566,-0.001690701,0.024555328,0.005717127
2016-03-10,-0.00393391,0.000494462,-0.002837383,NaN,-0.006364562,0.001017442,0.001740812,-0.004925531,0.003458948,-0.008042152,...,0.017772942,0.030652281,NaN,-0.002669903,-0.006028834,-0.060681629,0.006028834,0.004781829,-0.013827009,-0.013593673


In [9]:
data.tail()

,A,AAPL,ABBV,ABNB,ABT,ACGL,ACN,ADBE,ADI,ADM,...,WY,WYNN,XEL,XOM,XYL,XYZ,YUM,ZBH,ZBRA,ZTS
Date,,,,,,,,,,,,,,,,,,,,,
2026-03-05,-0.020159,-0.008495,-0.016258,0.002287,-0.018214,-0.013717,0.019048,0.031561,-0.034523,-0.010265,...,-0.007655,0.019743,-0.007948,0.006274,-0.024807,0.032802,-0.002327,-0.024021,-0.007562,-0.029613
2026-03-06,-0.025739,-0.010872,-0.009641,-0.019286,-0.013329,0.001442,0.004673,0.006673,-0.042187,0.013678,...,-0.005278,-0.035684,0.001699,0.002985,-0.024091,-0.015583,0.0087,0.007063,-0.043628,-0.006545
2026-03-07,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
2026-03-08,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
2026-03-09,-0.00504,-0.003535,-0.020773,-0.017113,0.019533,-0.016459,-0.039209,-0.012376,-0.012033,-0.003114,...,-0.012449,-0.024407,-0.011028,0.00086,-0.012505,-0.028343,-0.009375,-0.015726,-0.048486,-0.0175


In [10]:
import pandas_market_calendars as mcal

calendars = mcal.get_calendar("NYSE")
business_days = calendars.schedule(start_date = '2016-03-09', end_date = '2026-03-09')
df = data[data.index.isin(business_days.index)]

# Analyzing sources of missingness: (mainly just dropping at first)

In [11]:
# print(df.isna().mean().to_string())
# print(df.isna().sum()[df.isna().sum() > 0])
df = df[df.isna().sum()[df.isna().sum() == 0].index]

In [12]:
# print(df.isna().mean().to_string())
print(df.shape)
# df.columns = df.columns.astype(str)
df.to_hdf("Data/daily_returns_SPX.h5", key = "returns")

(2514, 428)


- Now we have 2514 daily returns over the last 10 years of 418 US equities.